lib/AST/DeclContext.cpp

```diff
@@ -271,6 +272,37 @@ ModuleDecl *DeclContext::getParentModule() const {
   return const_cast<ModuleDecl *>(cast<ModuleDecl>(DC));
 }
 
+/// Find the import that makes the given nominal declaration available.
+Optional<AttributedImport<ImportedModule>>
+DeclContext::findImportFor(ModuleDecl *nominalModule) const {
+  // If the nominal type is from the current module, there's no import.
+  if (nominalModule == getParentModule())
+    return None;
```

Comment:
Nit: `nominalModule` should probably renamed to `module` now that we're dealing with an arbitrary module input, not just the parent of a nominal type. Could you also tweak the comment not to say nominal type? Something like "If this is the current module, there's no import".